src/utils/logging.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export interface LogInterface {
+    debug(msg: string, ...supportingData: any[]): void;
+    warn(msg: string, ...supportingData: any[]): void;
+    error(msg: string, ...supportingData: any[]): void;
+    info(msg: string, ...supportingData: any[]): void;
+}
+
+export enum LogLevel {
+    debug,
+    info,
+    warn,
+    error
+}
+
+
+export class Log implements LogInterface {
+    public currentLogLevel = LogLevel.info;
+    private exitonerror = true;
+
+    constructor(loglevel: LogLevel, exitonerror: boolean) {
```

Comment:
I suggest adding `name` for the logger, this will also us to print the peripheral name before each log message, and in the future also configure different log names according to the peripheral name.

also, in the constructor, you can use `private readonly logLevel: LogLevel` to declare an initialize the property (instead of declaring it once in the class and once in the constructor and having to initialize it here).

Reply:
Done